pandas/core/ops.py

```diff
@@ -461,7 +461,7 @@ def _convert_to_array(self, values, name=None, other=None):
 
             # a datelike
             elif isinstance(values, pd.DatetimeIndex):
-                values = values.to_series()
```

Comment:
we should handle ``is_datetime64_dtype`` and ``is_datetime64tz_dtype`` separately here